### PR TITLE
Remove the filestore section from ceph-volume

### PR DESCRIPTION
**Pull request by fmount**
Filestore is no longer supported in cephadm and both the doc [1] and the DriveGroupValidation [2] raise an exception if this method is used. This patch removes the legacy code that is supposed to produce filestore ceph-volume related commands.

[1] https://github.com/ceph/ceph/blob/main/doc/cephadm/adoption.rst#limitations
[2] https://github.com/ceph/ceph/blob/main/src/python-common/ceph/deployment/drive_group.py#L366-L369

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by fmount** [comment]
@adk3798 @guits this is a follow up of #49555 and it's basically a code cleanup against `filestore` commands.

File: src/python-common/ceph/deployment/translate.py [hunk 1]
```diff
@@ -116,73 +116,53 @@ def run(self):
...
-            devs: List = sum(list(devices.values()), [])
-
-            cmd = "lvm batch --no-auto"
-
-            cmd += " {}".format(" ".join(devs))
-
-            if self.spec.journal_size:
-                cmd += " --journal-size {}".format(self.spec.journal_size)
-
-            if journal_devices:
```
Comment (adk3798) [inline src/python-common/ceph/deployment/translate.py:132]:
```
ceph/deployment/translate.py:98:9: F841 local variable 'journal_devices' is assigned to but never used
```

Comment (fmount) [reply src/python-common/ceph/deployment/translate.py:132]: Sorry, missed that line: updated in the current patchset.

**Comment by fmount** [comment]
jenkins test make check

**Comment by fmount** [comment]
jenkins test make check

**Comment by fmount** [comment]
jenkins test make check

**Comment by adk3798** [comment]
https://pulpito.ceph.com/adking-2023-03-06_16:09:09-orch:cephadm-wip-adk-testing-2023-03-04-1231-distro-default-smithi/

Failed/dead jobs reruns: https://pulpito.ceph.com/adking-2023-03-06_21:04:02-orch:cephadm-wip-adk-testing-2023-03-04-1231-distro-default-smithi/

After reruns, 6 failed, 2 dead jobs:

- 2 dead jobs were teuthology/infra issues (`Could not get lock /var/lib/dpkg/lock-frontend`, `SSH connection to smithi038 was lost`)
- 1 job failed installing a kernel `[Errno 2] No such file or directory: '/var/cache/dnf/CentOS-BaseOS-7c3d8b2bb640713c/packages/kernel-4.18.0-448.el8.x86_64.rpm'` unrelated to our tests
- 2 instances of https://tracker.ceph.com/issues/58535
- 3 jobs failed due to changes made to `ceph nfs cluster create` command in another PR in the run

Overall, should only block merging for PRs related to nfs

**Review by adk3798 — APPROVED** [review]

**Review by guits — APPROVED** [review]

**Pull request merged**